submissions/kt-std/friends-app/js/script.js

```diff
@@ -0,0 +1,447 @@
+            break;
+        case "namesAscending":
+            FRIENDS_ARRAY.sort((a, b) => a.firstName.localeCompare(b.firstName));
+            break;
+        case "ageDescending":
+            FRIENDS_ARRAY.sort((a, b) => b.age - a.age);
+            break;
+        case "ageAscending":
+            FRIENDS_ARRAY.sort((a, b) => a.age - b.age);
+            break;
+    }
+}
+
+function findSubstring(string, substring) {
+    return string.toLowerCase().indexOf(substring.toLowerCase()) >= 0;
+}
+
+function findMatchesWithPropertiesValues(
+    propertiesList,
+    friendsArray,
+    substring
+) {
+    return friendsArray.filter((friend) => {
+        return propertiesList
+            .map((property) => findSubstring(friend[property], substring))
+            .some((el) => el);
+    });
+}
+
+function filterByGender(e, friendsArray) {
+    let checkboxes = document.querySelectorAll("input[type=checkbox]");
+    checkboxes = Array.from(checkboxes).filter((checkbox) => checkbox.checked);
+    return friendsArray.filter((friend) =>
+        checkboxes.some((gender) => friend.gender === gender.value)
+    );
+}
+
+function filterByAge(friendsArray) {
+    const min = document.getElementById(MIN_AGE_INPUT_ID),
+        max = document.getElementById(MAX_AGE_INPUT_ID);
+    if (min.value && max.value) {
+        return friendsArray.filter(
+            (friend) => friend.age >= min.value && friend.age <= max.value
+        );
+    }
+}
+
+function checkFiltersChanged(event) {
+    return ["list__item", "number", "checkbox"].some(
+        (filter) => event.target.className === filter
+    );
+}
+
+function updateCards(event) {
+    if (checkFiltersChanged(event)) {
+        FRIENDS_ARRAY = INITIAL_FRIENDS_ARRAY;
+        sortCards(event, FRIENDS_ARRAY);
+        FRIENDS_ARRAY = filterByAge(FRIENDS_ARRAY);
+        FRIENDS_ARRAY = filterByGender(event, FRIENDS_ARRAY);
+        appendFriendsCards(FRIENDS_ARRAY);
+    }
+}
+
+function checkOptionsSelected() {
+    return SELECT_CONTAINER.attributes["option-selected"].value === "true";
+}
+
+function checkOptionsVisibility() {
+    return OPTIONS_CONTAINER.classList.contains("visible");
+}
+
+function resetOptionTabindex(option) {
+    option.tabIndex = -1;
+    option.removeAttribute("aria-selected");
+}
+
+function getSelectOption(selectText) {
+    return Array.from(OPTIONS_CONTAINER.children).filter(
+        (option) => option.textContent === selectText)[0];
+}
+
+function higlightSelectedOption(itemToSelect) {
+    itemToSelect.tabIndex = 0;
+    setTimeout(() => {
+        itemToSelect.focus();
+    }, 100);
+    itemToSelect.setAttribute("aria-selected", "true");
+    updateSelectText(itemToSelect.textContent);
+}
+
+function focusOnItem(buttonPressed) {
+    const selectedItem = document.activeElement;
+    if (buttonPressed === "ArrowUp") {
+        switch (true) {
+            case selectedItem.previousElementSibling &&
```

Comment:
You relying on DOM structure. Which is bad. What will be if layout will change? Your code will be broken